### ISSUE TITLE
Python 3.10.1 ImportError: DLL load failed while importing unicodedata: The specified module could not be found.

### ISSUE THREAD

**Issue by geostergiop** [mhostetter/galois#211]
Hi Matt,

Just updated python to 3.10.1 and previous scripts that use galois are not running. Galois Error thrown is `ImportError: DLL load failed while importing unicodedata: The specified module could not be found.`. 
Tried uninstalling/reinstalling galois with `pip3 uninstall galois` and `pip3 install git+https://github.com/mhostetter/galois.git` but to no avail.

Do you have any insights on this?
Thanks,
George

**Comment by mhostetter** [comment]
Hey, George. Python 3.10 isn't supported yet. The issue is with a dependency of `galois`, Numba. Numba doesn't yet support Python 3.10 and therefore neither can we.

I'm working on this in https://github.com/mhostetter/galois/pull/191. Currently installing the package doesn't work because there's no valid Numba version that is compatible with Python 3.10. I'm curious how you installed the package actually...

I just re-ran the PR for 3.10 and got this error at install.

```
Run python3 -m pip install .
Processing /home/runner/work/galois/galois
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'done'
Collecting numpy<1.21,>=1.18.4
  Downloading numpy-1.20.3.zip (7.8 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
ERROR: Could not find a version that satisfies the requirement numba<0.55,>=0.53 (from galois) (from versions: 0.1, 0.2, 0.3, 0.5.0, 0.6.0, 0.7.0, 0.7.1, 0.7.2, 0.8.0, 0.8.1, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.12.2, 0.13.0, 0.13.2, 0.13.3, 0.13.4, 0.14.0, 0.15.1, 0.16.0, 0.17.0, 0.18.1, 0.18.2, 0.19.1, 0.19.2, 0.20.0, 0.21.0, 0.22.0, 0.22.1, 0.23.0, 0.23.1, 0.24.0, 0.25.0, 0.26.0, 0.27.0, 0.28.1, 0.29.0, 0.30.0, 0.30.1, 0.31.0, 0.32.0, 0.33.0, 0.34.0, 0.35.0, 0.36.1, 0.36.2, 0.37.0, 0.38.0, 0.38.1, 0.39.0, 0.40.0, 0.40.1, 0.41.0, 0.42.0, 0.42.1, 0.43.0, 0.43.1, 0.44.0, 0.44.1, 0.45.0, 0.45.1, 0.46.0, 0.47.0, 0.48.0, 0.49.0, 0.49.1rc1, 0.49.1, 0.50.0rc1, 0.50.0, 0.50.1, 0.51.0rc1, 0.51.0, 0.51.1, 0.51.2, 0.52.0rc2, 0.55.0rc1)
ERROR: No matching distribution found for numba<0.55,>=0.53
```

The good news is Numba 0.55 will support Python 3.10 and they just released a release candidate (https://github.com/numba/numba/milestone/56). I suspect in January or so they'll release 0.55 and we'll be able to support Python 3.10.

**Comment by geostergiop** [comment]
ΟΚ nice, sorry for the duplicate with #191, did a quick pass but didn't get it. I ll close this issue.

**Comment by mhostetter** [comment]
I suppose if you really wanted to be cutting edge, you could install the Numba 0.55rc and test if it's compatible with `galois`. You'll be the first to find out, so there may be some bugs...

To do it, do the following:

Install the latest Numba from `master`
```
$ pip3 uninstall numba
$ pip3 install git+https://github.com/numba/numba.git
```

Install my `python-3.10` branch of `galois`
```
$ pip3 uninstall galois
$ pip3 install git+https://github.com/mhostetter/galois.git@python-3.10
```

**Comment by geostergiop** [comment]
Thanks, no need to play with fire to be honest, I just reverted to 3.8 :-)

**Comment by mhostetter** [comment]
@geostergiop, Python 3.10 support was just added and merged into `master` with #191! :+1:
